pydicom/tests/test_config.py

```diff
@@ -98,10 +102,45 @@ def test_debug_off_handler_stream(self, caplog):
         assert isinstance(self.logger.handlers[0], logging.NullHandler)
         assert isinstance(self.logger.handlers[1], logging.StreamHandler)
 
-        with caplog.at_level(logging.DEBUG, logger='pydicom'):
+        with caplog.at_level(logging.DEBUG, logger="pydicom"):
             ds = dcmread(DS_PATH)
 
             assert "Call to dcmread()" not in caplog.text
             assert "Reading File Meta Information preamble..." in caplog.text
             assert "Reading File Meta Information prefix..." in caplog.text
             assert "00000080: 'DICM' prefix found" in caplog.text
+
+
+@pytest.fixture(scope="function", params=["config", "env"])
+def future_setter(request, monkeypatch):
+    if request.param == "config":
+        config.future_behavior()
+    else:
+        monkeypatch.setenv("PYDICOM_FUTURE", "True")
+        importlib.reload(config)
+
+
+class TestFuture:
+    def teardown(self):
+        # shouldn't need this, but monkeypatch env wasn't resetting
+        config._use_future = False
```

Comment:
You can move this into the fixture:
```py
def future_setter(request, monkeypatch):
       ... # setup
    yield
    config._use.future = False

Reply:
Yes... I had originally was going to do that with importlib.reload() but thought the env would still be set so that wouldn't work... was trying to avoid direct use of `_use_future` but it seemed I had to anyway.

I think it probably best to keep it all together, though, so will update as per your suggestion.

Reply:
It's actually strange that `monkeypatch` does not reset the environment variable - in should do this... But it does not do a reload of course, so the variable will not be reset.

Reply:
Done.